R/extract-transform-sherlock.R

```diff
@@ -158,28 +174,76 @@ process_plate_layout <- function(filepath, plate_size) {
 #' * plate_run_id
 #' @export
 #' @md
-process_well_sample_details <- function(filepath, sample_type, assay_type, plate_run_id) {
+process_well_sample_details <- function(filepath, sample_type, layout_type, single_assay_type, plate_run_id) {
 
+  layout_type <- tolower(layout_type)
   sample_type_id <- if_else(sample_type == "mucus", 1, 2)
 
-  assay_id <- case_when(assay_type == "Ots28_Early1" ~ 1,
-                        assay_type == "Ots28_Late1" ~ 2,
-                        assay_type == "Ots16_Spring1" ~ 3,
-                        assay_type == "Ots16_Winter1" ~ 4)
-
   layout_raw <- suppressMessages(read_excel(filepath,
                                             sheet = "Plate Map"))
+  # split plate
+  if(str_detect(layout_type, "split_plate")) {
+
+    assay_ids <- case_when(layout_type == "split_plate_early_late" ~ c(1, 2),
+                           layout_type == "split_plate_spring_winter" ~ c(3, 4))
+
+    plate_layout <- layout_raw |>
+      pivot_longer(names_to="col_num", values_to = "sample_id", -...1) |>
+      rename(row_num = ...1) |>
+      mutate(assay_id = ifelse(col_num %in% 1:12, assay_ids[1], assay_ids[2])) |>
```

Comment:
just a double check here that assay_id == 1 is "early" and assay_id == 2 is "late"